### PR TITLE
feat(iam): workspace one and nsx manager/edge password management

### PR DESCRIPTION
**Pull request by GaryJBlake**
In order to have a good experience with our community, we recommend that you read the [contributing guidelines](https://github.com/vmware-samples/power-validated-solutions-for-cloud-foundation/blob/main/CONTRIBUTING.md) for making a pull request.

**Summary of Pull Request**

- Added `Request-WsaPasswordExpiration` cmdlet to retrieve the password expiration policy for Workspace ONE Access.
- Added `Request-WsaPasswordComplexity` cmdlet to retrieve the password complexity policy for Workspace ONE Access.
- Added `Request-WsaAccountLockout` cmdlet to retrieve the account lockout policy for Workspace ONE Access.
- Added `Update-WsaPasswordExpiration` cmdlet to update the password expiration policy for Workspace ONE Access.
- Added `Update-WsaPasswordComplexity` cmdlet to update the password complexity policy for Workspace ONE Access.
- Added `Update-WsaAccountLockout` cmdlet to update the account lockout policy for Workspace ONE Access.
- Added `Request-NsxtManagerAccountLockout` cmdlet to retrieve the account lockout policy for NSX Manager nodes.
- Added `Update-NsxtManagerAccountLockout` cmdlet to update the account lockout policy for NSX Manager nodes.
- Added `Request-NsxtEdgeAccountLockout` cmdlet to retrieve the account lockout policy for NSX Edge nodes.
- Added `Update-NsxtEdgeAccountLockout` cmdlet to update the account lockout policy for NSX Edge nodes.
- Renamed `Get-WSAPasswordLockout` to `Get-WsaAccountLockout`
- Renamed `Set-WSAPasswordLockout` to `Set-WsaAccountLockout`
- Enhanced `Set-WsaPasswordPolicy` cmdlet to improve the output from the API.

Signed-off-by: Gary Blake <gblake@vmware.com>

**Type of Pull Request**

- [ ] This is a bug fix.
- [x] This is an enhancement or feature.
- [ ] This is a code style / formatting update.
- [ ] This is a documentation update.
- [ ] This is a refactoring update.
- [ ] This is something else.

**Related to Existing Issues**

Issue Number: N/A

**Test and Documentation Coverage**

- [x] Tests have been completed (for bug fixes / features).
- [ ] Documentation has been added / updated (for bug fixes / features).

**Breaking Changes?**

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### CODE REVIEW

File: PowerValidatedSolutions.psm1 [hunk 1]
```diff
@@ -16869,12 +16870,300 @@ Export-ModuleMember -Function Update-VcenterRootPasswordExpiration
...
+    <#
+		.SYNOPSIS
+        Retrieve account lockout policy from NSX Manager Nodes
+
+        .DESCRIPTION
+        The Request-NsxtManagerAccountLockout cmdlet retrieves the account lockout policy for each NSX Manager node for
+        a workload domain. The cmdlet connects to SDDC Manager using the -server, -user, and -password values:
+        - Validates that network connectivity and authentication is possible to SDDC Manager
+        - Validates that network connectivity and authentication is possible to NSX Manager
+        - Retrieves the account lockpout policy
```
[the remark this answers is not in the document]
Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:16883]: Fixed

File: PowerValidatedSolutions.psm1 [hunk 2]
```diff
@@ -16869,12 +16870,300 @@ Export-ModuleMember -Function Update-VcenterRootPasswordExpiration
...
+        .DESCRIPTION
+        The Request-NsxtEdgeAccountLockout cmdlet retrieves the account lockout policy from NSX Edge nodes within a
+        workload domain. The cmdlet connects to SDDC Manager using the -server, -user, and -password values:
+        - Validates that network connectivity and authentication is possible to SDDC Manager
+        - Validates that network connectivity and authentication is possible to NSX Management Cluster
+        - Retrieves the account lockpout policy
+
+        .EXAMPLE
+        Request-NsxtEdgeAccountLockout -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01
+        This example retrieving the account lockout policy for NSX Edge nodes in sfo-m01 workload domain
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:17044]:
```suggestion
        This example retrieves the account lockout policy for NSX Edge nodes in sfo-m01 workload domain
```

Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:17044]: Fixed

File: PowerValidatedSolutions.psm1 [hunk 3]
```diff
@@ -16869,12 +16870,300 @@ Export-ModuleMember -Function Update-VcenterRootPasswordExpiration
...
+        Configure account lockout policy for NSX Edge Nodes
+
+        .DESCRIPTION
+        The Update-NsxtEdgeAccountLockout cmdlet configures the account lockout policy for NSX Edge nodes.
+        The cmdlet connects to SDDC Manager using the -server, -user, and -password values:
+        - Validates that network connectivity and authentication is possible to SDDC Manager
+        - Validates that network connectivity and authentication is possible to NSX Management Cluster
+        - Configure the account lockout policy
+
+        .EXAMPLE
```
[the remark this answers is not in the document]
Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:17102]: We could of course, it is more an internal switch for use with PlatformTools

[the remark this answers is not in the document]
Comment (tenthirtyam) [reply PowerValidatedSolutions.psm1:17102]: Gotcha - let's leave it as-is here.

File: PowerValidatedSolutions.psm1 [hunk 4]
```diff
@@ -16887,6 +17176,290 @@ Export-ModuleMember -Function Update-VcenterRootPasswordExpiration
...
+		if (Test-WsaConnection -server $server) {
+			if (Test-WsaAuthentication -server $server -user $user -pass $pass) {
+                if ($WsaAccountLockout = Get-WsaAccountLockout) {
+                    $WsaAccountLockoutObject = New-Object -TypeName psobject
+                    $WsaAccountLockoutObject | Add-Member -notepropertyname "FQDN" -notepropertyvalue $server
+                    $WsaAccountLockoutObject | Add-Member -notepropertyname "Max Failures" -notepropertyvalue $WsaAccountLockout.numAttempts
+                    $WsaAccountLockoutObject | Add-Member -notepropertyname "Unlock Interval (min)" -notepropertyvalue $WsaAccountLockout.unlockInterval
+                    $WsaAccountLockoutObject | Add-Member -notepropertyname "Failed Attempt Interval (min)" -notepropertyvalue $WsaAccountLockout.attemptInterval                   
+                } else {
+                    Write-Error "Unable to retrieve account lockout policy from Workspace ONE Access instance ($server): PRE_VALIDATION_FAILED"
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:17299]: Suggestion to follow the established format:

```suggestion
                    Write-Error "Unable to retrieve Account Lockout Policy from Workspace ONE Access instance ($server): PRE_VALIDATION_FAILED"
```

Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:17299]: Fixed

File: PowerValidatedSolutions.psm1 [hunk 5]
```diff
@@ -16887,6 +17176,290 @@ Export-ModuleMember -Function Update-VcenterRootPasswordExpiration
...
+
+	)
+
+	Try {
+        if (Test-WsaConnection -server $server) {
+            if (Test-WsaAuthentication -server $server -user $user -pass $pass) {
+                $failureInterval = ($failureInterval / 60)
+                $unlockInterval = ($unlockInterval / 60)
+                if ((Get-WsaAccountLockout).numAttempts -ne $failures -or (Get-WsaAccountLockout).attemptInterval -ne $failureInterval -or (Get-WsaAccountLockout).unlockInterval -ne $unlockInterval) {
+                    Set-WsaAccountLockout  -numAttempts $failures -attemptInterval $failureInterval -unlockInterval $unlockInterval | Out-Null
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:17446]: Extra space:

```suggestion
                    Set-WsaAccountLockout -numAttempts $failures -attemptInterval $failureInterval -unlockInterval $unlockInterval | Out-Null
```

**Review by tenthirtyam — APPROVED** [review]
LGTM! 🚀

**Pull request merged**